sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -596,4 +646,54 @@ private void finishProcessingLabMessage(LabMessageDto labMessageDto) {
 		FacadeProvider.getLabMessageFacade().save(labMessageDto);
 		SormasUI.get().getNavigator().navigateTo(LabMessagesView.VIEW_NAME);
 	}
+
+	private void showAlreadyProcessedPopup(Component component) {
+		VerticalLayout warningLayout = VaadinUiUtil.createWarningLayout();
+		Window popupWindow = VaadinUiUtil.showPopupWindow(warningLayout);
+		Label infoLabel = new Label(I18nProperties.getValidationError(Validations.labMessageAlreadyProcessedError));
+		CssStyles.style(infoLabel, CssStyles.LABEL_LARGE, CssStyles.LABEL_WHITE_SPACE_NORMAL);
+		warningLayout.addComponent(infoLabel);
+		popupWindow.addCloseListener(e -> popupWindow.close());
+		popupWindow.setWidth(400, Sizeable.Unit.PIXELS);
+
+		// If something was saved by the user,
+		if (entityCreated) {
+			Button button = createDeleteEntityButton(component);
+			button.addClickListener(e -> popupWindow.close());
+			warningLayout.addComponent(button);
+		}
+	}
+
+	private Button createDeleteEntityButton(Component component) {
+		if (SampleCreateForm.class.equals(component.getClass())) {
```

Comment:
In principal I agree. In this case, designing this method in a more stable way (_move null check into method and handling out of sync component and entityCreated_) will result in wrong SORMAS behaviour without exception (_including the possibility to delete non new created entities_). That is my reasoning behind this unstable design.
Is that convincing to you?

Reply:
I think I understand the reasoning that this will signal if something is not correctly setup, however at least it should be in the method documentation that we don't expect null here and also maybe for `showAlreadyProcessedPopup` that if `component` is null and `entityCreated` true, it might result in a NPE.